File: src/main/java/toker/panel/entity/IpRecord.java
package toker.panel.entity;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "ip_record")
public class IpRecord {

    private Integer id;
    private Integer uniqueId;
    private String ipAddress;
    private Server server;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    @Column(name = "unique_id", nullable = false)
    public Integer getUniqueId() {
        return uniqueId;
    }

    public void setUniqueId(Integer uniqueId) {
        this.uniqueId = uniqueId;
    }

    @Column(name = "ip_address", nullable = false, length = 16)
    public String getIpAddress() {
        return ipAddress;
    }

    public void setIpAddress(String ipAddress) {
        this.ipAddress = ipAddress;
    }

    @ManyToOne
    @JoinColumn(name = "server_id", referencedColumnName = "id", nullable = false)
    public Server getServer() {
        return server;
    }

    public void setServer(Server server) {
        this.server = server;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        IpRecord ipRecord = (IpRecord) o;
        return Objects.equals(uniqueId, ipRecord.uniqueId) &&
                Objects.equals(ipAddress, ipRecord.ipAddress);
    }

    @Override
    public int hashCode() {
        return Objects.hash(uniqueId, ipAddress);
    }
}

File: src/main/java/toker/panel/entity/IpRecord.kt
package toker.panel.entity

import javax.persistence.*

@Entity
@Table(name = "ip_record")
class IpRecord(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        @Column(name = "id")
        var id: Int? = null,

        @Column(name = "unique_id", nullable = false)
        var uniqueId: Int? = null,

        @Column(name = "ip_address", nullable = false, length = 16)
        var ipAddress: String? = null,

        @ManyToOne
        @JoinColumn(name = "server_id", referencedColumnName = "id", nullable = false)
        var server: Server? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as IpRecord
        if (id != other.id) return false
        return true
    }

    override fun hashCode(): Int {
        return id ?: 0
    }
}